### ISSUE TITLE
Add rule block-opening-brace-empty-line-after

### ISSUE THREAD

**Issue by doochik** [stylelint/stylelint#4366]
> What is the problem you're trying to solve?

I want to have consistent code in this cases

```css

.a {
  /* empty line after open brace */
  color: red;
}

.a { /* no empty line after open brace */
  color: red;
}
```

> What solution would you like to see?

Add rule `block-opening-brace-empty-line-after`. This is symmetric rule to `block-closing-brace-empty-line-before`

I can create PR with this rule.

**Comment by hudochenkov** [comment]
Both of these cases doesn't have an empty line after opening brace. If I understood correctly, you want [block-opening-brace-newline-after](https://stylelint.io/user-guide/rules/block-opening-brace-newline-after), but which doesn't ignore first comment on the same line as brace.

**Comment by doochik** [comment]
> Both of these cases doesn't have an empty line after opening brace. If I understood correctly, you want [block-opening-brace-newline-after](https://stylelint.io/user-guide/rules/block-opening-brace-newline-after), but which doesn't ignore first comment on the same line as brace.

No, I insert comments to illustrate problem :)
I want to remove all empty lines after opening brace (no rule for this) and before closing brace (`block-closing-brace-empty-line-before`)

```css

/** This line
     ↓ */
a /* ↓ */
{ /* ↓ */
  /* ← */
  color: pink;
}
```

Examples for 'never'
```css
/* bad for me */
.a {
  
  color: red;
}

/* good for me */
.a {
  color: red;
}
```

**Comment by doochik** [comment]
@hudochenkov what do you think about this rule?

**Comment by hudochenkov** [comment]
I think this rule has intersection with other rules. Check this [demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICGACYAdAdllLIBsIAneFAYgDEqBuLAXyxABoQBnAT0wBckAPBcK1bNwETADMAlgHMBGTLnQgiAV3wxWS0vNyKQsMPiREkXSWIC0MALYAHLuwv5JmGBYBGMccRhaUAbRxdPSR8AHckdk1mIOCdYN0lGF4wGHs-QIUE4KUkcS4YIgsDIxMzMSUmWOyUJSkiVi4LV0aYKCVq4IBdKqzspRlMHwzOhNz8wotIa2sYbkrRnJAXVklYC1XMaXUnFzd3QjAAaw6+hK7RhjOLrKu6EDogA). Configured `declaration-empty-line-before` already shows an error for empty line after opening bracket.

These are the rules, which would confilct with proposed rule:

- `at-rule-empty-line-before`
- `comment-empty-line-before`
- `custom-property-empty-line-before`
- `declaration-empty-line-before`
- `rule-empty-line-before`

I don't think we need `block-opening-brace-empty-line-after`, because use case for preventing empty line is covered by other rules.

**Comment by doochik** [comment]
`declaration-empty-line-before` has conflict with plugin `order/properties-order`.

config example:
```
'order/properties-order': [
            [
                {
                    emptyLineBefore: 'always',
                    noEmptyLineBetween: true,
                    properties: [
                        'font-size',
                    ],
                },
                {
                    emptyLineBefore: 'always',
                    noEmptyLineBetween: true,
                    properties: [
                        'position',
                    ],
                },
```

code example
```css
.a {
  font-size: 10px;

  position: absolute;
}
```

**Comment by hudochenkov** [comment]
`declaration-empty-line-before` should not be used if `emptyLineBefore` is used in `order/properties-order`. Issue related to your problem already opened in https://github.com/hudochenkov/stylelint-order/issues/95.

**Comment by doochik** [comment]
Yes, I there is not rule to prevent me to write like this

```css
.a {

  font-size: 10px;

  position: absolute;
}
```

Empty line after opening brace is valid

**Comment by hudochenkov** [comment]
Core rules could be conflicting with third-party rules, but core rules shouldn't be conflicting with core rules. We're trying to avoid conflicts between rules, so it won't be possible to create configurations, which will produce confusing results. That's why we have the following in our [Criteria for inclusion](https://stylelint.io/developer-guide/rules#criteria-for-inclusion):

> Does not contain functionality that overlaps with another rule.

I'm thinking that proposed rule shouldn't be part of stylelint core, because it overlaps with other rules, as I pointed out in https://github.com/stylelint/stylelint/issues/4366#issuecomment-544537887.

@stylelint/contributors what do you think?

@doochik you did a great job in https://github.com/stylelint/stylelint/pull/4367. It's always possible to [create a plugin](https://stylelint.io/developer-guide/plugins).

**Comment by doochik** [comment]
Ok, I'll create the plugin

**Comment by alkorlos** [comment]
@doochik I had the same problem. `declaration-empty-line-before: never` fixed it when stylelint was 9.8.0, now it doesn’t fix.
I agree that the rule `block-opening-brace-empty-line-after` would be useful, thak you for the [plagin](https://www.npmjs.com/package/stylelint-plugin-block-opening-brace-empty-line-after)
